contributors/devel/contributing-api-changes.md

```diff
@@ -0,0 +1,134 @@
+# Guidelines for contributing api changes
+
+## Overview
+
+Modifications to the core Kubernetes api have a material impact on the Kubernetes project, and
+undergo thorough review and discussion before they are accepted.  This document outlines the
+review process and what is expected of api contributors.
+
+## Changes that qualify as api changes
+
+Any changes to existing or new apis are subject to the process outlined in this document.
+
+The following qualify as api changes:
+
+- Adding, removing, or modifying api types (e.g. through types.go, or by defining a new annotation)
+- Changing the version of an existing type (e.g. v1alpha1 to v1beta1)
+- Adding, removing, or modifying subresources for existing types. (e.g. /scale)
+- Material changes to how fields on existing types are interpreted / used
+
+The following on their own do not qualify as api changes, but should follow the process defined by the related
+[special interest groups].
+
+- Adding or changing a command or flag to kubectl: sig-cli
+- Building new command line tools: sig-cli
+- Adding or changing a flag to kubelet: sig-node
+- Adding or changing a flag to apiserver: sig-api-machinery
+- Refactoring code
+- Building extensions
+
+## Life of an api change
+
+1. Start a discussion
+2. Write a short description
+3. Write a detailed design proposal
+4. Schedule a design review
+5. Implement the proposal
+
+### Start a discussion.
+
+Expected time for decision: 1-4 weeks (depending on the sig and feature)
+
+Start a discussion with the relevant [special interest groups] that will [own] the changes.
+Depending on the sig, this may be done via sig meetings, email groups, slack channel, etc.
+
+### Write a short description
+
+Expected time for decision: 2 weeks
+
+Write a short description of the api changes by copying the [design summary template]
+into a new issue, and [@mention] @kubernetes/api-reviewers and the owning sig reviewers
+group.
+
+Determine whether @kubernetes/api-reviewers is willing to accept a design proposal
+for your feature.  The answer could be "no, we don't want to solve this problem this way."
+or "no, this is something we want, but cannot commit the resources to reviewing it this release."
+
+### Write a detailed design proposal
+
+Expected time for initial feedback: 3-6 weeks
+
+Write a detailed design proposal PR using the [design proposal PR template].  [@mention]
+@kubernetes/api-reviewers and the appropriate sigs on the PR and reference the original
+issue in the PR description.
+
+### Schedule a design review
+
+Expected time for next slot: 4-8 weeks
+
+Schedule a time slow for your design to be reviewed.
+
+Design approvers:
```

Comment:
My current thinking is that we need to find a way to allow more rapid experimentation for new APIs that would be alpha, especially if they aren't in the kubernetes repo, then do a more rigorous review with the senior (bottleneck) API approvers before the API advances to beta, which is when we commit to the API, guarantee backward compatibility, etc.